src/components/Tables/Filters.tsx

```diff
@@ -1,36 +1,35 @@
 import React from "react";
 import { TableHeaders } from "@/components/Tables/Table";
+import { Database } from "@/databaseTypesFile";
+import { PostgrestFilterBuilder } from "@supabase/postgrest-js";
+
+export enum PaginationType {
+    Server = "SERVER",
+    Client = "CLIENT",
+}
+
+export type MethodConfig<Data, State> =
+    | {
+          method: (
+              query: PostgrestFilterBuilder<Database["public"], any, any>,
+              state: State
+          ) => PostgrestFilterBuilder<Database["public"], any, any>;
+          methodType: PaginationType.Server;
+      }
+    | {
+          method: (row: Data, state: State, key: keyof Data) => boolean;
+          methodType: PaginationType.Client;
```

Comment:
some tables require client side filtering (admin tables, and the component tests).